core/src/main/java/greencity/converters/UserArgumentResolver.java

```diff
@@ -41,6 +42,11 @@ public boolean supportsParameter(MethodParameter parameter) {
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
         NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         Principal principal = webRequest.getUserPrincipal();
-        return principal != null ? userService.findByEmail(principal.getName()) : null;
+
+        if (principal == null) {
+            throw new UnauthorizedException("Unauthorized access");
```

Comment:
As you always use the same `Unauthorized access` message it can be moved to the `UnauthorizedException` class